File: 350-intersection-of-two-arrays-ii/350-intersection-of-two-arrays-ii.cpp
class Solution {
public:
    vector<int> intersect(vector<int>& nums1, vector<int>& nums2) {
        
        vector<int> crossnums;
        
        sort(nums1.begin(), nums1.end());
        sort(nums2.begin(), nums2.end());
        
        int i = nums1.size() - 1;
        int j = nums2.size() - 1;
        
        while(i>=0 && j>=0){
            if(nums1[i] == nums2[j]){
                crossnums.push_back(nums1[i]);
                i -= 1;
                j -= 1;
            }
            else if(nums1[i] < nums2[j]){
                j -= 1;
            }
            else{
                i -= 1;
            }
        }
        
        return crossnums;
        }
};

File: 350-intersection-of-two-arrays-ii/350-intersection-of-two-arrays-ii.py
class Solution:
    def intersect(self, nums1: List[int], nums2: List[int]) -> List[int]:
        hash_nums1 = {}
        hash_nums2 = {}
        hash_intersection = {}
        arr = []
        
        for num in nums1:
            if num in hash_nums1:
                hash_nums1[num] += 1
            else:
                hash_nums1[num] = 1
        
        for num in nums2:
            if num in hash_nums2:
                hash_nums2[num] += 1
            else:
                hash_nums2[num] = 1
        
        for key in hash_nums1.keys():
            if key in hash_nums2:
                hash_intersection[key] = min(hash_nums1[key], hash_nums2[key])
        
        for key, value in hash_intersection.items():
            arr += [key for x in range(value)]
        
        return arr